presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/TestMongoIntegrationSmokeTest.java

```diff
@@ -43,13 +42,13 @@
 
     public TestMongoIntegrationSmokeTest()
     {
-        super(() -> createMongoQueryRunner(ORDERS));
+        this(createMongoQueryRunner(ORDERS));
```

Comment:
this eagerly starts everything

* test static init and ctors are very sensitive to failure, because maven surefire does not properly report exceptions happening during that phase
* we allocate all memory (threads, dockers) upfront even if we run tests single-threaded or in 2 threads

the code will be way easier to write if you update this class to override `AbstractTestQueryFramework#createQueryRunner` instead of providing the supplier in the super constructor.